projects/onto-search/src/lib/onto-search-results/onto-search-results.component.ts

```diff
@@ -1,21 +1,40 @@
-import {Component, Input, OnChanges, SimpleChanges} from '@angular/core';
+import {Component, EventEmitter, Input, OnChanges, Output, SimpleChanges} from '@angular/core';
 import {TableConfig} from './models/configuration-types';
+import {Sort} from '@angular/material/sort';
 
 @Component({
   selector: 'onto-search-results',
   templateUrl: './onto-search-results.component.html',
   styleUrls: ['./onto-search-results.component.scss'],
 })
 export class OntoSearchResultsComponent implements OnChanges {
+  /**
+   * The data source for the table
+   */
   @Input()
   public datasource: any[];
 
+  /**
+   * Table configuration, containing table and column specific configuration
+   * If column configuration is changed dynamically the whole columnConfigurations
+   * array must be overridden for changes to be detected
```

Comment:
You mean that a new instance of the array should be created or what?

Reply:
Yes. But I will change that.
I am thinking about implementing DoCheck doing the diff checks manually.
This way there will be no need for that.

Reply:
But why do you consider changing it? I kind of prefer the current approach.

Reply:
ok

Reply:
I have decided to leave it as is. I have clarified the comment.